content/blog/2024-07-31-valkey-8-0-0-rc1.md

```diff
@@ -0,0 +1,131 @@
+Valkey 8.0 improves reliability and minimizes disruptions with the following
+enhancements:
+
+- **Automatic Failover for Empty Shards**: New shards that start empty, owning
+  no slots, now benefit from automatic failover. This ensures high availability
+  from the start of the scaling process.
+- **Replication of Slot Migration States**: All `CLUSTER SETSLOT` commands are
+  now replicated synchronously to replicas before execution on the primary. This
+  reduces the chance of unavailability if the primary fails, as the replicas have
+  the most up-to-date information about the state of the shard. New Replicas also
+  automatically inherit the state from the primary without additional input from
+  an operator.
+- **Logical Link Recovery**: In the event of a failover, Valkey 8.0 automatically
+  repairs the logical links between source and target nodes. This ensures requests
+  are correctly routed to the new primary in the target shard, maintaining cluster
+  integrity and availability.
+
+## Replication
+
+Valkey 8.0 introduces a dual-channel replication scheme, allowing the RDB and
+the replica backlog to be transferred simultaneously, accelerating synchronization.
+
+- **Reduced Memory Load**: By streaming replication data to the replica during
+  the full sync, the primary node experiences significantly less memory pressure.
+  The replica now manages the Client Output Buffer (COB) tracking, reducing the
+  likelihood of COB overruns and enabling larger COB sizes on the replica side.
+- **Reduced Parent Process Load**: A dedicated connection for RDB transfer frees
+  the primary's parent process from handling this data, allowing it to focus on
+  client queries and improving overall responsiveness.
+
+Performance tests show improvements in write latency during sync, and in scenarios
+with heavy read commands, the sync time can be cut by up to 50%. This translates
+to a more responsive system, even during synchronization.
+
+## Observability
+
+Valkey 8.0 introduces a comprehensive per-slot metrics infrastructure, providing
+detailed visibility into the performance and resource usage of individual slots.
+This granular data helps inform decisions about resource allocation, load
+balancing, and performance optimization.
+
+- **Key Count**: Returns the number of keys in each slot, making it easier to identify the slots with the largest number of keys.
+- **CPU Usage**: Tracks CPU time consumed by operations on each slot, identifying
+  areas of high utilization and potential bottlenecks.
+- **Network Input/Output Bytes**: Monitors data transmission and reception by
+  each slot, offering insights into network load and bandwidth utilization.
+- **Minimal Overhead**: Initial benchmarks show that enabling detailed metrics
+  incurs a negligible overhead of approximately 0.7% in QPS.
+
+## Efficiency
+
+Valkey 8.0 introduces key embedding in the main dictionary, eliminating separate
```

Comment:
```suggestion
Valkey 8.0 introduces two new improvements that reduce the memory overhead of keys, allowing users to store more data without any application changes.
The change change is that keys are now embedded in the main dictionary, eliminating separate
```
This section felt like it should have more of an introduction, similar to the other sections.